numba/dispatcher.py

```diff
@@ -395,30 +395,75 @@ def error_rewrite(e, issue_type):
             raise e
 
     def inspect_llvm(self, signature=None):
+        """Get the LLVM intermediate representation generated by compilation.
+
+        Parameters
+        ----------
+        signature : tuple of numba types, optional
+            Specify a signature for which to obtain the LLVM IR. If None, the
+            IR is returned for all available signatures.
+
+        Returns
+        -------
+        llvm : dict[signature, str] or str
+            Either the LLVM IR string for the specified signature, or, if no
+            signature was given, a dictionary mapping signatures to LLVM IR
+            strings.
+        """
         if signature is not None:
             lib = self.overloads[signature].library
             return lib.get_llvm_str()
 
         return dict((sig, self.inspect_llvm(sig)) for sig in self.signatures)
 
     def inspect_asm(self, signature=None):
+        """Get the generated human-readable native assembler code.
+
+        Parameters
+        ----------
+        signature : tuple of numba types, optional
+            Specify a signature for which to obtain the assembler code. If
+            None, the code is returned for all available signatures.
+
+        Returns
+        -------
+        asm : dict[signature, str] or str
+            Either the assembler for the specified signature, or, if no
+            signature was given, a dictionary mapping signatures to assembler
+            code.
+        """
         if signature is not None:
             lib = self.overloads[signature].library
             return lib.get_asm_str()
 
         return dict((sig, self.inspect_asm(sig)) for sig in self.signatures)
 
-    def inspect_types(self, file=None, signature=None, **kwargs):
-        """
-        print or return annotated source with Numba intermediate IR
+    def inspect_types(self, file=None, signature=None,
+                      pretty=False, style='default', **kwargs):
+        """Print/return code annotated with Numba intermediate representation.
 
-        Pass `pretty=True` to attempt color highlighting, and HTML rendering in
-        Jupyter and IPython by returning an Annotate Object. `file` must be
-        None if used in conjunction with `pretty=True`.
+        Parameters
+        ----------
+        file : file-like object, optional
+            File to which to print. Defaults to sys.stdout if None. Must be
+            None if ``pretty=True``.
+        signature : tuple of numba types, optional
+            Print/return intermediate representation for only the given
+            signature. If None, the IR is printed for all available signatures.
+        pretty : bool, optional
+            If True, an Annotate object will be returned that can render the
+            IR with color highlighting in Jupyter and IPython. ``file`` must
+            be None if ``pretty`` is True.
+        style : str, optional
+            Choose a style for rendering.
```

Comment:
Perhaps mention the style is WRT the pygments formatter classes?

Reply:
@stuartarchibald this might have been easier to write if there had been any existing indication anywhere about this. =P I still don't actually know what "WRT the pygments formatter classes" actually means... Any doc links? What are the options here? Totally into improving this. =)

Reply:
Indeed, TBH I recalled this from memory only as I have a recollection of working on the backend stuff for the patches that made this work :)

Drilling down through the code, seems like `style` is consumed by the `Annotate` ctor here:
https://github.com/numba/numba/blob/81b91f93c60cd4bfe9d6c1cfce1191e64bdf2a5c/numba/dispatcher.py#L439

passed to this:
https://github.com/numba/numba/blob/81b91f93c60cd4bfe9d6c1cfce1191e64bdf2a5c/numba/pretty_annotate.py#L224-L282

and subsequently consumed here https://github.com/numba/numba/blob/81b91f93c60cd4bfe9d6c1cfce1191e64bdf2a5c/numba/pretty_annotate.py#L10-L21 by the `HTMLFormatter` class from Pygments, docs for that are here: http://pygments.org/docs/formatters/#HtmlFormatter

Hope this helps?

Reply:
Indeed, TBH I recalled this from memory only as I have a recollection of working on the backend stuff for the patches that made this work :)

Drilling down through the code, seems like `style` is consumed by the `Annotate` ctor here:
https://github.com/numba/numba/blob/81b91f93c60cd4bfe9d6c1cfce1191e64bdf2a5c/numba/dispatcher.py#L439

passed to this:
https://github.com/numba/numba/blob/81b91f93c60cd4bfe9d6c1cfce1191e64bdf2a5c/numba/pretty_annotate.py#L224-L282

and subsequently consumed here https://github.com/numba/numba/blob/81b91f93c60cd4bfe9d6c1cfce1191e64bdf2a5c/numba/pretty_annotate.py#L10-L21 by the `HTMLFormatter` class from Pygments, docs for that are here: http://pygments.org/docs/formatters/#HtmlFormatter

Hope this helps?